junifer/preprocess/warping/space_warper.py

```diff
@@ -0,0 +1,198 @@
+            The list of data types that can be used as input for this
+            preprocessor.
+
+        """
+        return ["T1w", "BOLD", "VBM_GM", "VBM_WM", "fALFF", "GCOR", "LCOR"]
+
+    def get_output_type(self, input_type: str) -> str:
+        """Get output type.
+
+        Parameters
+        ----------
+        input_type : str
+            The data type input to the preprocessor.
+
+        Returns
+        -------
+        str
+            The data type output by the preprocessor.
+
+        """
+        # Does not add any new keys
+        return input_type
+
+    def preprocess(
+        self,
+        input: Dict[str, Any],
+        extra_input: Optional[Dict[str, Any]] = None,
+    ) -> Tuple[Dict[str, Any], Optional[Dict[str, Dict[str, Any]]]]:
+        """Preprocess.
+
+        Parameters
+        ----------
+        input : dict
+            The input from the Junifer Data object.
+        extra_input : dict, optional
+            The other fields in the Junifer Data object.
+
+        Returns
+        -------
+        dict
+            The computed result as dictionary.
+        None
+            Extra "helper" data types as dictionary to add to the Junifer Data
+            object.
+
+        Raises
+        ------
+        ValueError
+            If ``extra_input`` is None when transforming to native space
+            i.e., using ``"T1w"`` as reference.
+        RuntimeError
+            If the data is in the correct space and does not require
+            warping.
+
+        """
+        logger.info(f"Warping to {self.reference} space using SpaceWarper")
+        # Transform to native space
+        if self.reference == "T1w":
+            # Check for extra inputs
+            if extra_input is None:
+                raise_error(
+                    "No extra input provided, requires `Warp` and "
+                    f"`{self.reference}` data types in particular."
+                )
+            # Conditional preprocessor
+            if self.using == "fsl_native":
+                input = FSLNativeWarper().preprocess(
+                    input=input,
+                    extra_input=extra_input,
+                )
+            elif self.using == "ants_native":
+                input = ANTsNativeWarper().preprocess(
+                    input=input,
+                    extra_input=extra_input,
+                )
+        # Transform to template space
+        else:
+            # Check pre-requirements for space manipulation
+            if self.reference == input["space"]:
+                raise_error(
+                    (
+                        "Skipped warping as the data is in "
+                        f"{self.reference} space which would mean that you "
+                        "can remove the SpaceWarper from the preprocess step."
+                    ),
+                    klass=RuntimeError,
+                )
+            else:
+                input = ANTsTemplateWarper().preprocess(
```

Comment:
This also has a conflict with the `using` parameter. It does not consider the `using` here, so `fsl` or `ants` is the same here.

Reply:
Unfortunately, there is no other current approach for warping templates in junifer so having a forced check for `using` doesn't really make sense. Of course we can make it but it doesn't give anything extra in return. I can make the base `if...else` stronger if that eases this out.